src/generators/schema/author.php

```diff
@@ -127,7 +127,13 @@ protected function determine_user_id( Meta_Tags_Context $context ) {
 		 *
 		 * @api int|bool $user_id The user ID currently determined.
 		 */
-		return apply_filters( 'wpseo_schema_person_user_id', $user_id );
+		$user_id = \apply_filters( 'wpseo_schema_person_user_id', $user_id );
+
+		if ( \is_int( $user_id ) && $user_id > 0 ) {
+			return $user_id;
+		}
+
+		return false;
```

Comment:
I would return 0 here, because the code that implements this method already checks if the value it falsy. This way you have one return type for this method. 

Another thing you can consider is throwing an exception and catch this (but that might be a little bit overkill)

Reply:
See https://github.com/Yoast/wordpress-seo/pull/14620/files#diff-e14ce197b7f0a09bad85c14e3c7cae06L91